### ISSUE TITLE
Windows apps hang opening files in \\wsl$ or \\wsl.localhost

### ISSUE THREAD

**Issue by nathdolan** [microsoft/WSL#7479]
### Windows Build Number

Microsoft Windows [Version 10.0.22000.194]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.10.43

### Distro Version

Ubuntu 18.04

### Repro Steps

1. Open notepad (or any other Windows application)
2. Attempt to open any file from anywhere in the Ubuntu folder (\\wsl$ or \\wsl.localhost)

### Expected Behavior

The file should open!  

### Actual Behavior

Notepad or any other application hangs for approx 2 minutes 30 seconds before finally opening the file

### Diagnostic Logs

This was working fine a short while ago, but I can't pinpoint when this started hapenning.

Recently I:
- Upgraded from current Windows build to Insider Beta to try out Windows 11
- Tried out wslg
- Experimented with Docker Desktop for Windows which did not work well so reverted back to using dockerd within Ubuntu.

Weirdly accessing folders in the  \\wsl$ or \\wsl.localhost filesystems via Explorer is fine, but not files.

I only had \\wsl$, not \\wsl.localhost before.  I guess wsl.localhost is a new thing?  Should I have both?

In Ubuntu I configure my profile to mount windows drives (/mnt/c, /mnt/d etc) to root (/c, /d etc).  This never caused any problems before but thought I'd mention it.

Everything works fine within WSL2/Ubuntu.

I am guessing this is a network request being blocked by something, loopback timeout or something but can't pinpoint where it is coming from.

**Comment by korpa** [comment]
I have the same issue

```
WSL-Version: 0.50.2.0
Kernelversion: 5.10.74.3
WSLg-Version: 1.0.29
Windows-Version: 10.0.22000.318
```

Ubuntu 20.04

After I reboot Windows access works fine for a couple of minutes to a couple of hours but then hangs. Restarting WSL via `wsl --shutdown` and open WSL again does not fix the issue. I have to reboot.

**Comment by Jytesh** [comment]
I have this Same issue 
```
WSL version: 0.50.2.0
Kernel version: 5.10.74.3
WSLg version: 1.0.29
Windows version: 10.0.22000.376
````

I can open explorer.exe ( and all others ) from Ubuntu when I am in /mnt/c, but not when im in any other ubuntu/wsl folders

Attempting to click on Ubuntu in the Explorer is unresponsive, only the command line and VSCode - Remote WSL are able to access wsl folders

**Comment by Jytesh** [comment]
I found https://github.com/microsoft/WSL/issues/7782 so I would recommend closing all these other issues of the same problem

**Comment by DanPinGF** [comment]
> Hi! We've identified this issue as a duplicate of another one that already exists in this repository. This specific instance is being closed in favor of tracking the concern over on the referenced thread.
> 
> Thanks for your report!

Um... this doesn't make any sense. Why is this being closed as a duplicate of an issue we're currently on right now?

**Comment by webprogrammierer** [comment]
I have the same problem:

Notepad++ or any other editor (notepad.exe ...) hangs up for 5 to 10 minutes before the file is opened in the editor when I try to open a file in Windows from one of these wsl pathes:
- \\wsl.localhost\
- \\wsl$

Important detail:
Files with the following file extension open without problems: .txt, .md, .config, .yml, .scss
Files with the following files extensions can not be opened in the editor: .php, .json, .htaccess, .js, .css. The editor hangs for minutes.

php. .js ..... files that can be executed and could be dangerous   -> Windows is blocking these files 
.txt, .md, .yml .... text files -> no danger, Windows does not block these files

A few days ago everything was fine and this problem did not exist.

Anybody here who already solved this problem without reinstalling WSL2?
What can I do to get rid of the hang ups when opening files with the specific file endings (php, js, json, ...)?

Thank you very much.

**Comment by webprogrammierer** [comment]
I solved the problem by myself by adding these 3 Folders
\\wsl.localhost\Ubuntu
\\wsl.localhost\docker-desktop
\\wsl.localhost\docker-desktop-data
to the excluded folders of Avira Security (settings).
